middleware/routeHandler.js

```diff
@@ -1,10 +1,28 @@
+const { getCommitAndTreeSha, revertCommit } = require('../utils/utils.js')
+
 const attachRouteHandlerWrapper = (routeHandler) => async (req, res, next) => {
-    routeHandler(req, res).catch((err) => {
-      next(err)
-    })
+  routeHandler(req, res).catch((err) => {
+    next(err)
+  })
+}
+
+const attachRollbackRouteHandlerWrapper = (routeHandler) => async (req, res, next) => {
+  const { accessToken } = req
+  const { siteName } = req.params
+  let originalCommitSha
+  try {
+    const { currentCommitSha } = await getCommitAndTreeSha(siteName, accessToken)
+    originalCommitSha = currentCommitSha
+  } catch (err) {
+    next(err)
   }
+  routeHandler(req, res).catch((err) => {
+    revertCommit(originalCommitSha, siteName, accessToken)
```

Comment:
We might need a retry mechanism for `revertCommit` if it fails

Reply:
As discussed, we can retry this with exponential backoff. Resolved in https://github.com/isomerpages/isomercms-backend/pull/91/commits/004d904fb79cd59dda82937d0e1b1a41d8483d2b